src/contentmanager.cpp

```diff
@@ -501,6 +508,19 @@ void ContentManager::openBook(const QString &id)
     }
 }
 
+void ContentManager::openBookPreview(const QString &id)
+{
+    const auto& book = getRemoteOrLocalBook(id);
+
+    std::string pathStr = book.getUrl();
+
+    auto pathStartIndex = pathStr.find_last_of('/') + 1;
+    pathStr = "https://library.kiwix.org/viewer#" +
+              pathStr.substr(pathStartIndex,
+                             pathStr.find(".zim", pathStartIndex) - pathStartIndex);
```

Comment:
The preview URL has the form `BASEURL/viewer#BOOKNAME` where `BOOKNAME` can be obtained via `kiwix::Book::getName()` (instead of parsing/deducing it from the result of `kiwix::Book::getUrl()`).

`BASEURL` can be [overriden while debugging](https://github.com/kiwix/kiwix-desktop/commit/c10026a3a2616245f5ea47dedf3be436f5e07bc1). Please generate `BASEURL` as [this](https://github.com/kiwix/kiwix-desktop/commit/535675afdb0fce40121ccafa06909b61ad057895#diff-b143bce41389bc342b560b90019ea25d3a82243fab36352befb2be7401cc2b64R682-R701). In order to avoid introducing code duplication, please add a private helper function `ContentManager::getRemoteLibraryUrl()`.

Reply:
My previous comment regarding the `BOOKNAME` part of the preview URL was wrong. Actually [ZIM name](https://kiwix-tools.readthedocs.io/en/latest/kiwix-serve.html#term-ZIM-name) is used (see https://kiwix-tools.readthedocs.io/en/latest/kiwix-serve.html#viewer). Note that ZIM name may be different from the ZIM filename (without the extension), however I think that ZIM filenames for content provided by Kiwix are safe to be used as ZIM names. So your code for deducing the ZIM name from the download URL should work. But please compute it in a separate expression:

```c++
const auto zimname = ...;
const auto previewUrl = getRemoteLibraryUrl() + "/viewer#" + zimname;
```

Reply:
Done